src/main/java/fi/helsinki/cs/tmc/core/commands/DownloadAdaptiveExercise.java

```diff
@@ -0,0 +1,40 @@
+package fi.helsinki.cs.tmc.core.commands;
+
+import fi.helsinki.cs.tmc.core.communication.TmcServerCommunicationTaskFactory;
+import fi.helsinki.cs.tmc.core.domain.Exercise;
+import fi.helsinki.cs.tmc.core.domain.Progress;
+import fi.helsinki.cs.tmc.core.domain.ProgressObserver;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DownloadAdaptiveExercise extends ExerciseDownloadingCommand<Exercise> {
+
+    private static final Logger logger = LoggerFactory.getLogger(DownloadAdaptiveExercise.class);
+
+    public DownloadAdaptiveExercise(ProgressObserver observer) {
+        super(observer);
+    }
+
+    @VisibleForTesting
+    DownloadAdaptiveExercise(ProgressObserver observer, TmcServerCommunicationTaskFactory tmcServerCommunicationTaskFactory) {
+        super(observer, tmcServerCommunicationTaskFactory);
+    }
+
+    @Override
+    public Exercise call() throws Exception {
+        logger.info("Checking adaptive exercises availability");
+        Exercise exercise = tmcServerCommunicationTaskFactory.getAdaptiveExercise().call();
+        if (exercise == null) {
+            return null;
+        }
+        exercise.setName("AdaLovelace");
+        exercise.setCourseName("None");
```

Comment:
This is hard to fix for this pull request, because we have built the sprint 1 on top of this, which uses this method. We have made a task for this, and will fix it asap in this sprint.